src/recipe.rs

```diff
@@ -460,10 +466,8 @@ impl<'src, D> Recipe<'src, D> {
   }
 
   pub(crate) fn doc(&self) -> Option<&str> {
-    for attribute in &self.attributes {
-      if let Attribute::Doc(doc) = attribute {
-        return doc.as_ref().map(|s| s.cooked.as_ref());
-      }
+    if let Some(Attribute::Doc(doc)) = self.attributes.get(AttributeDiscriminant::Doc) {
```

Comment:
This is kind of a weird one. This `if let` will only match if *both* `self.attributes.get` returns `Some`, and the returned attribute is an `Attribute::Doc`, if the code is correct, of course, both or neither will be the case.

I think a more scrupulous correct version would be:

```
if let Some(attribute) = self.attributes.get(AttributeDiscriminant::Doc) {
  if let Some(Attribute::Doc(doc)) else {
    unreachable!();
  }
}
```

Although that's a little crazy, so I think maybe the original on the left is better.